transport/peer.go

```diff
@@ -0,0 +1,48 @@
+package transport
+
+//go:generate mockgen -destination=transporttest/peer.go -package=transporttest go.uber.org/yarpc/transport PeerIdentifier,Peer
+
+// PeerConnectionStatus maintains information about the Peer's connection state
+type PeerConnectionStatus int
+
+const (
+	// PeerUnavailable indicates the Peer is unavailable for requests
+	PeerUnavailable PeerConnectionStatus = iota
+
+	// PeerConnecting indicates the Peer is in the process of connecting
+	PeerConnecting
+
+	// PeerAvailable indicates the Peer is available for requests
+	PeerAvailable
```

Comment:
Yeah, makes sense, unfortunately with the stacked diffs changing it right now might be a bit painful, but changing it at the end of the stack should be NBD #424